src/clj/game/cards/identities.clj

```diff
@@ -505,10 +505,18 @@
              :msg "make the Runner spend 1 [Credits] to access"}]})
 
 (defcard "GameNET: Where Dreams are Real"
-  {:implementation "Credit gain not implemented. You can use shortcut ability."
-   :abilities [{:msg "gain 1 [Credits] (shortcut)"
-                :async true
-                :effect (effect (gain-credits :corp eid 1))}]})
+  (let [gamenet-ability {:req (req (and (:run @state)
+                                        (= (:side (:source eid)) "Corp")
+                                        (or (and (not= (:source-type eid) :runner-trash-corp-cards)
+                                                 (not= (:source-type eid) :runner-steal))
+                                            (some (fn [[cost source]] (and (some #(or (= (first %) :credit) (= (first %) :x-credits)) (merge-costs cost))
```

Comment:
```suggestion
                                            (some (fn [[cost source]] (and (some #(or (= (cost-name %) :credit) (= (cost-name %) :x-credits)) (merge-costs cost))
```

Better to use the multi-methods for accessing cost related stuff, just in case we ever decide to change how they're handled.